tests/unit/api/objectPutACL.js

```diff
@@ -462,30 +462,38 @@ describe('putObjectACL API', () => {
         });
     });
 
-    it('should set originOp to s3:ObjectAcl:Put when ACL is changed', done => {
-        const testObjACLRequest = {
-            bucketName,
-            namespace,
-            objectKey: objectName,
-            headers: { 'x-amz-acl': 'public-read-write' },
-            url: `/${bucketName}/${objectName}?acl`,
-            query: { acl: '' },
-        };
-
-        bucketPut(authInfo, testPutBucketRequest, log, () => {
-            objectPut(authInfo, testPutObjectRequest, undefined, log,
-                (err, resHeaders) => {
-                    assert.strictEqual(resHeaders.ETag, `"${correctMD5}"`);
-                    objectPutACL(authInfo, testObjACLRequest, log, err => {
-                        assert.strictEqual(err, null);
-                        metadata.getObjectMD(bucketName, objectName, {},
-                        log, (err, md) => {
-                            assert.strictEqual(md.originOp,
-                            's3:ObjectAcl:Put');
-                            done();
+    [
+        { headers: { 'x-amz-acl': 'public-read-write' }, type: 'ACL' },
+        { headers: { 'x-amz-grant-full-control': 'emailaddress="sampleaccount1@sampling.com"' }, type: 'FULL_CONTROL' },
+        { headers: { 'x-amz-grant-read': `uri=${constants.logId}` }, type: 'READ' },
+        { headers: { 'x-amz-grant-read-acp': `id=${ownerID}` }, type: 'READ_ACP' },
+        { headers: { 'x-amz-grant-write-acp': `id=${anotherID}` }, type: 'WRITE_ACP' },
+    ].forEach(params => {
```

Comment:
don't we need to also check combinations of grants ?

note that maybe it would be easier to test the various conditions/paths by calling `addObjectACL` directly (while keeping a separate test for the objectPutACL API itself, to check if the API does the correct parsing/plumbing/error handling/... ?